### PR TITLE
SLVS-1063 Change analysis-output.xml to a streamable protobuf file

### PR DESCRIPTION
**Pull request by tvsonar**

### CODE REVIEW

File: src/SonarAnalyzer.Scanner/Protobuf/AnalyzerReport.cs [hunk 1]
```diff
@@ -0,0 +1,1835 @@
...
+      int size = 0;
+      if (FilePath.Length != 0) {
+        size += 1 + pb::CodedOutputStream.ComputeStringSize(FilePath);
+      }
+      size += reference_.CalculateSize(_repeated_reference_codec);
+      return size;
+    }
+
+    public void MergeFrom(SymbolReferenceInfo other) {
+      if (other == null) {
```
Comment (SonarTech) [inline src/SonarAnalyzer.Scanner/Protobuf/AnalyzerReport.cs]: MAJOR In member 'SymbolReferenceInfo.MergeFrom(SymbolReferenceInfo)', change parameter name 'other' to 'message' in order to match the identifier as it has been declared in 'IMessage<SymbolReferenceInfo>.MergeFrom(SymbolReferenceInfo)'. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=fxcop%3AParameterNamesShouldMatchBaseDeclaration)

File: src/SonarAnalyzer.Scanner/Protobuf/AnalyzerReport.cs [hunk 2]
```diff
@@ -0,0 +1,1835 @@
...
+    private int endOffset_;
+    public int EndOffset {
+      get { return endOffset_; }
+      set {
+        endOffset_ = value;
+      }
+    }
+
+    public override bool Equals(object other) {
+      return Equals(other as TextRange);
```
Comment (SonarTech) [inline src/SonarAnalyzer.Scanner/Protobuf/AnalyzerReport.cs]: MAJOR In member 'TextRange.Equals(object)', change parameter name 'other' to 'obj' in order to match the identifier as it has been declared in 'object.Equals(object)'. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=fxcop%3AParameterNamesShouldMatchBaseDeclaration)

File: src/SonarAnalyzer.Scanner/Protobuf/AnalyzerReport.cs [hunk 3]
```diff
@@ -0,0 +1,1835 @@
...
+      int size = 0;
+      if (FilePath.Length != 0) {
+        size += 1 + pb::CodedOutputStream.ComputeStringSize(FilePath);
+      }
+      size += tokenInfo_.CalculateSize(_repeated_tokenInfo_codec);
+      return size;
+    }
+
+    public void MergeFrom(CopyPasteTokenInfo other) {
+      if (other == null) {
```
Comment (SonarTech) [inline src/SonarAnalyzer.Scanner/Protobuf/AnalyzerReport.cs]: MAJOR In member 'CopyPasteTokenInfo.MergeFrom(CopyPasteTokenInfo)', change parameter name 'other' to 'message' in order to match the identifier as it has been declared in 'IMessage<CopyPasteTokenInfo>.MergeFrom(CopyPasteTokenInfo)'. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=fxcop%3AParameterNamesShouldMatchBaseDeclaration)

File: src/SonarAnalyzer.Scanner/Protobuf/AnalyzerReport.cs [hunk 4]
```diff
@@ -0,0 +1,1835 @@
...
+        private global::SonarAnalyzer.Protobuf.TextRange textRange_;
+        public global::SonarAnalyzer.Protobuf.TextRange TextRange {
+          get { return textRange_; }
+          set {
+            textRange_ = value;
+          }
+        }
+
+        public override bool Equals(object other) {
+          return Equals(other as TokenInfo);
```
Comment (SonarTech) [inline src/SonarAnalyzer.Scanner/Protobuf/AnalyzerReport.cs]: MAJOR In member 'TokenTypeInfo.Types.TokenInfo.Equals(object)', change parameter name 'other' to 'obj' in order to match the identifier as it has been declared in 'object.Equals(object)'. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=fxcop%3AParameterNamesShouldMatchBaseDeclaration)

File: src/SonarAnalyzer.Scanner/Protobuf/AnalyzerReport.cs [hunk 5]
```diff
@@ -0,0 +1,1835 @@
...
+            size += 1 + pb::CodedOutputStream.ComputeStringSize(TokenValue);
+          }
+          if (textRange_ != null) {
+            size += 1 + pb::CodedOutputStream.ComputeMessageSize(TextRange);
+          }
+          return size;
+        }
+
+        public void MergeFrom(TokenInfo other) {
+          if (other == null) {
```
Comment (SonarTech) [inline src/SonarAnalyzer.Scanner/Protobuf/AnalyzerReport.cs]: MAJOR In member 'CopyPasteTokenInfo.Types.TokenInfo.MergeFrom(CopyPasteTokenInfo.Types.TokenInfo)', change parameter name 'other' to 'message' in order to match the identifier as it has been declared in 'IMessage<CopyPasteTokenInfo.Types.TokenInfo>.MergeFrom(CopyPasteTokenInfo.Types.TokenInfo)'. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=fxcop%3AParameterNamesShouldMatchBaseDeclaration)

File: src/SonarAnalyzer.Scanner/Protobuf/AnalyzerReport.cs [hunk 6]
```diff
@@ -0,0 +1,1835 @@
...
+        size += 1 + pb::CodedOutputStream.ComputeInt32Size(StartOffset);
+      }
+      if (EndOffset != 0) {
+        size += 1 + pb::CodedOutputStream.ComputeInt32Size(EndOffset);
+      }
+      return size;
+    }
+
+    public void MergeFrom(TextRange other) {
+      if (other == null) {
```
Comment (SonarTech) [inline src/SonarAnalyzer.Scanner/Protobuf/AnalyzerReport.cs]: MAJOR In member 'TextRange.MergeFrom(TextRange)', change parameter name 'other' to 'message' in order to match the identifier as it has been declared in 'IMessage<TextRange>.MergeFrom(TextRange)'. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=fxcop%3AParameterNamesShouldMatchBaseDeclaration)

File: src/SonarAnalyzer.Scanner/Protobuf/AnalyzerReport.cs [hunk 7]
```diff
@@ -0,0 +1,1835 @@
...
+    public const int TokenInfoFieldNumber = 2;
+    private static readonly pb::FieldCodec<global::SonarAnalyzer.Protobuf.TokenTypeInfo.Types.TokenInfo> _repeated_tokenInfo_codec
+        = pb::FieldCodec.ForMessage(18, global::SonarAnalyzer.Protobuf.TokenTypeInfo.Types.TokenInfo.Parser);
+    private readonly pbc::RepeatedField<global::SonarAnalyzer.Protobuf.TokenTypeInfo.Types.TokenInfo> tokenInfo_ = new pbc::RepeatedField<global::SonarAnalyzer.Protobuf.TokenTypeInfo.Types.TokenInfo>();
+    public pbc::RepeatedField<global::SonarAnalyzer.Protobuf.TokenTypeInfo.Types.TokenInfo> TokenInfo {
+      get { return tokenInfo_; }
+    }
+
+    public override bool Equals(object other) {
+      return Equals(other as TokenTypeInfo);
```
Comment (SonarTech) [inline src/SonarAnalyzer.Scanner/Protobuf/AnalyzerReport.cs]: MAJOR In member 'TokenTypeInfo.Equals(object)', change parameter name 'other' to 'obj' in order to match the identifier as it has been declared in 'object.Equals(object)'. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=fxcop%3AParameterNamesShouldMatchBaseDeclaration)

File: src/SonarAnalyzer.Scanner/Protobuf/AnalyzerReport.cs [hunk 8]
```diff
@@ -0,0 +1,1835 @@
...
+        size += 1 + pb::CodedOutputStream.ComputeStringSize(FunctionComplexityDistribution);
+      }
+      size += noSonarComment_.CalculateSize(_repeated_noSonarComment_codec);
+      size += nonBlankComment_.CalculateSize(_repeated_nonBlankComment_codec);
+      size += codeLine_.CalculateSize(_repeated_codeLine_codec);
+      return size;
+    }
+
+    public void MergeFrom(MetricsInfo other) {
+      if (other == null) {
```
Comment (SonarTech) [inline src/SonarAnalyzer.Scanner/Protobuf/AnalyzerReport.cs]: MAJOR In member 'MetricsInfo.MergeFrom(MetricsInfo)', change parameter name 'other' to 'message' in order to match the identifier as it has been declared in 'IMessage<MetricsInfo>.MergeFrom(MetricsInfo)'. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=fxcop%3AParameterNamesShouldMatchBaseDeclaration)

File: src/SonarAnalyzer.Scanner/Protobuf/AnalyzerReport.cs [hunk 9]
```diff
@@ -0,0 +1,1835 @@
...
+        public const int ReferenceFieldNumber = 2;
+        private static readonly pb::FieldCodec<global::SonarAnalyzer.Protobuf.TextRange> _repeated_reference_codec
+            = pb::FieldCodec.ForMessage(18, global::SonarAnalyzer.Protobuf.TextRange.Parser);
+        private readonly pbc::RepeatedField<global::SonarAnalyzer.Protobuf.TextRange> reference_ = new pbc::RepeatedField<global::SonarAnalyzer.Protobuf.TextRange>();
+        public pbc::RepeatedField<global::SonarAnalyzer.Protobuf.TextRange> Reference {
+          get { return reference_; }
+        }
+
+        public override bool Equals(object other) {
+          return Equals(other as SymbolReference);
```
Comment (SonarTech) [inline src/SonarAnalyzer.Scanner/Protobuf/AnalyzerReport.cs]: MAJOR In member 'SymbolReferenceInfo.Types.SymbolReference.Equals(object)', change parameter name 'other' to 'obj' in order to match the identifier as it has been declared in 'object.Equals(object)'. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=fxcop%3AParameterNamesShouldMatchBaseDeclaration)

File: src/SonarAnalyzer.Scanner/Protobuf/AnalyzerReport.cs [hunk 10]
```diff
@@ -0,0 +1,1835 @@
...
+    public const int CodeLineFieldNumber = 15;
+    private static readonly pb::FieldCodec<int> _repeated_codeLine_codec
+        = pb::FieldCodec.ForInt32(122);
+    private readonly pbc::RepeatedField<int> codeLine_ = new pbc::RepeatedField<int>();
+    public pbc::RepeatedField<int> CodeLine {
+      get { return codeLine_; }
+    }
+
+    public override bool Equals(object other) {
+      return Equals(other as MetricsInfo);
```
Comment (SonarTech) [inline src/SonarAnalyzer.Scanner/Protobuf/AnalyzerReport.cs]: MAJOR In member 'MetricsInfo.Equals(object)', change parameter name 'other' to 'obj' in order to match the identifier as it has been declared in 'object.Equals(object)'. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=fxcop%3AParameterNamesShouldMatchBaseDeclaration)

File: src/SonarAnalyzer.Scanner/Protobuf/AnalyzerReport.cs [hunk 11]
```diff
@@ -0,0 +1,1835 @@
...
+    public const int IssueFieldNumber = 2;
+    private static readonly pb::FieldCodec<global::SonarAnalyzer.Protobuf.FileIssues.Types.Issue> _repeated_issue_codec
+        = pb::FieldCodec.ForMessage(18, global::SonarAnalyzer.Protobuf.FileIssues.Types.Issue.Parser);
+    private readonly pbc::RepeatedField<global::SonarAnalyzer.Protobuf.FileIssues.Types.Issue> issue_ = new pbc::RepeatedField<global::SonarAnalyzer.Protobuf.FileIssues.Types.Issue>();
+    public pbc::RepeatedField<global::SonarAnalyzer.Protobuf.FileIssues.Types.Issue> Issue {
+      get { return issue_; }
+    }
+
+    public override bool Equals(object other) {
+      return Equals(other as FileIssues);
```
Comment (SonarTech) [inline src/SonarAnalyzer.Scanner/Protobuf/AnalyzerReport.cs]: MAJOR In member 'FileIssues.Equals(object)', change parameter name 'other' to 'obj' in order to match the identifier as it has been declared in 'object.Equals(object)'. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=fxcop%3AParameterNamesShouldMatchBaseDeclaration)

File: src/SonarAnalyzer.Scanner/Protobuf/AnalyzerReport.cs [hunk 12]
```diff
@@ -0,0 +1,1835 @@
...
+      int size = 0;
+      if (FilePath.Length != 0) {
+        size += 1 + pb::CodedOutputStream.ComputeStringSize(FilePath);
+      }
+      size += tokenInfo_.CalculateSize(_repeated_tokenInfo_codec);
+      return size;
+    }
+
+    public void MergeFrom(TokenTypeInfo other) {
+      if (other == null) {
```
Comment (SonarTech) [inline src/SonarAnalyzer.Scanner/Protobuf/AnalyzerReport.cs]: MAJOR In member 'TokenTypeInfo.MergeFrom(TokenTypeInfo)', change parameter name 'other' to 'message' in order to match the identifier as it has been declared in 'IMessage<TokenTypeInfo>.MergeFrom(TokenTypeInfo)'. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=fxcop%3AParameterNamesShouldMatchBaseDeclaration)

File: src/SonarAnalyzer.Scanner/Protobuf/AnalyzerReport.cs [hunk 13]
```diff
@@ -0,0 +1,1835 @@
...
+    public const int ReferenceFieldNumber = 2;
+    private static readonly pb::FieldCodec<global::SonarAnalyzer.Protobuf.SymbolReferenceInfo.Types.SymbolReference> _repeated_reference_codec
+        = pb::FieldCodec.ForMessage(18, global::SonarAnalyzer.Protobuf.SymbolReferenceInfo.Types.SymbolReference.Parser);
+    private readonly pbc::RepeatedField<global::SonarAnalyzer.Protobuf.SymbolReferenceInfo.Types.SymbolReference> reference_ = new pbc::RepeatedField<global::SonarAnalyzer.Protobuf.SymbolReferenceInfo.Types.SymbolReference>();
+    public pbc::RepeatedField<global::SonarAnalyzer.Protobuf.SymbolReferenceInfo.Types.SymbolReference> Reference {
+      get { return reference_; }
+    }
+
+    public override bool Equals(object other) {
+      return Equals(other as SymbolReferenceInfo);
```
Comment (SonarTech) [inline src/SonarAnalyzer.Scanner/Protobuf/AnalyzerReport.cs]: MAJOR In member 'SymbolReferenceInfo.Equals(object)', change parameter name 'other' to 'obj' in order to match the identifier as it has been declared in 'object.Equals(object)'. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=fxcop%3AParameterNamesShouldMatchBaseDeclaration)

File: src/SonarAnalyzer.Scanner/Protobuf/AnalyzerReport.cs [hunk 14]
```diff
@@ -0,0 +1,1835 @@
...
+        private global::SonarAnalyzer.Protobuf.TextRange textRange_;
+        public global::SonarAnalyzer.Protobuf.TextRange TextRange {
+          get { return textRange_; }
+          set {
+            textRange_ = value;
+          }
+        }
+
+        public override bool Equals(object other) {
+          return Equals(other as TokenInfo);
```
Comment (SonarTech) [inline src/SonarAnalyzer.Scanner/Protobuf/AnalyzerReport.cs]: MAJOR In member 'CopyPasteTokenInfo.Types.TokenInfo.Equals(object)', change parameter name 'other' to 'obj' in order to match the identifier as it has been declared in 'object.Equals(object)'. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=fxcop%3AParameterNamesShouldMatchBaseDeclaration)

File: src/SonarAnalyzer.Scanner/Protobuf/AnalyzerReport.cs [hunk 15]
```diff
@@ -0,0 +1,1835 @@
...
+            size += 1 + pb::CodedOutputStream.ComputeStringSize(Message);
+          }
+          if (location_ != null) {
+            size += 1 + pb::CodedOutputStream.ComputeMessageSize(Location);
+          }
+          return size;
+        }
+
+        public void MergeFrom(Issue other) {
+          if (other == null) {
```
Comment (SonarTech) [inline src/SonarAnalyzer.Scanner/Protobuf/AnalyzerReport.cs]: MAJOR In member 'FileIssues.Types.Issue.MergeFrom(FileIssues.Types.Issue)', change parameter name 'other' to 'message' in order to match the identifier as it has been declared in 'IMessage<FileIssues.Types.Issue>.MergeFrom(FileIssues.Types.Issue)'. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=fxcop%3AParameterNamesShouldMatchBaseDeclaration)

File: src/SonarAnalyzer.Scanner/Protobuf/AnalyzerReport.cs [hunk 16]
```diff
@@ -0,0 +1,1835 @@
...
+            size += 1 + pb::CodedOutputStream.ComputeEnumSize((int) TokenType);
+          }
+          if (textRange_ != null) {
+            size += 1 + pb::CodedOutputStream.ComputeMessageSize(TextRange);
+          }
+          return size;
+        }
+
+        public void MergeFrom(TokenInfo other) {
+          if (other == null) {
```
Comment (SonarTech) [inline src/SonarAnalyzer.Scanner/Protobuf/AnalyzerReport.cs]: MAJOR In member 'TokenTypeInfo.Types.TokenInfo.MergeFrom(TokenTypeInfo.Types.TokenInfo)', change parameter name 'other' to 'message' in order to match the identifier as it has been declared in 'IMessage<TokenTypeInfo.Types.TokenInfo>.MergeFrom(TokenTypeInfo.Types.TokenInfo)'. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=fxcop%3AParameterNamesShouldMatchBaseDeclaration)

File: src/SonarAnalyzer.Scanner/Protobuf/AnalyzerReport.cs [hunk 17]
```diff
@@ -0,0 +1,1835 @@
...
+      }
+      noSonarComment_.Add(other.noSonarComment_);
+      nonBlankComment_.Add(other.nonBlankComment_);
+      codeLine_.Add(other.codeLine_);
+    }
+
+    public void MergeFrom(pb::CodedInputStream input) {
+      uint tag;
+      while ((tag = input.ReadTag()) != 0) {
+        switch(tag) {
```
Comment (SonarTech) [inline src/SonarAnalyzer.Scanner/Protobuf/AnalyzerReport.cs]: MAJOR 'MetricsInfo.MergeFrom(CodedInputStream)' has a cyclomatic complexity of 27. Rewrite or refactor the method to reduce complexity to 25. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=fxcop%3AAvoidExcessiveComplexity)

File: src/SonarAnalyzer.Scanner/Protobuf/AnalyzerReport.cs [hunk 18]
```diff
@@ -0,0 +1,1835 @@
...
+
+    #region Descriptor
+    /// <summary>File descriptor for AnalyzerReport.proto</summary>
+    public static pbr::FileDescriptor Descriptor {
+      get { return descriptor; }
+    }
+    private static pbr::FileDescriptor descriptor;
+
+    static AnalyzerReportReflection() {
+      byte[] descriptorData = global::System.Convert.FromBase64String(
```
Comment (SonarTech) [inline src/SonarAnalyzer.Scanner/Protobuf/AnalyzerReport.cs]: MAJOR Initialize all static fields in 'AnalyzerReportReflection' when those fields are declared and remove the explicit static constructor. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=fxcop%3AInitializeReferenceTypeStaticFieldsInline)

File: src/SonarAnalyzer.Scanner/Protobuf/AnalyzerReport.cs [hunk 19]
```diff
@@ -0,0 +1,1835 @@
...
+    public const int TokenInfoFieldNumber = 2;
+    private static readonly pb::FieldCodec<global::SonarAnalyzer.Protobuf.CopyPasteTokenInfo.Types.TokenInfo> _repeated_tokenInfo_codec
+        = pb::FieldCodec.ForMessage(18, global::SonarAnalyzer.Protobuf.CopyPasteTokenInfo.Types.TokenInfo.Parser);
+    private readonly pbc::RepeatedField<global::SonarAnalyzer.Protobuf.CopyPasteTokenInfo.Types.TokenInfo> tokenInfo_ = new pbc::RepeatedField<global::SonarAnalyzer.Protobuf.CopyPasteTokenInfo.Types.TokenInfo>();
+    public pbc::RepeatedField<global::SonarAnalyzer.Protobuf.CopyPasteTokenInfo.Types.TokenInfo> TokenInfo {
+      get { return tokenInfo_; }
+    }
+
+    public override bool Equals(object other) {
+      return Equals(other as CopyPasteTokenInfo);
```
Comment (SonarTech) [inline src/SonarAnalyzer.Scanner/Protobuf/AnalyzerReport.cs]: MAJOR In member 'CopyPasteTokenInfo.Equals(object)', change parameter name 'other' to 'obj' in order to match the identifier as it has been declared in 'object.Equals(object)'. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=fxcop%3AParameterNamesShouldMatchBaseDeclaration)

File: src/SonarAnalyzer.Scanner/Protobuf/AnalyzerReport.cs [hunk 20]
```diff
@@ -0,0 +1,1835 @@
...
+          int size = 0;
+          if (declaration_ != null) {
+            size += 1 + pb::CodedOutputStream.ComputeMessageSize(Declaration);
+          }
+          size += reference_.CalculateSize(_repeated_reference_codec);
+          return size;
+        }
+
+        public void MergeFrom(SymbolReference other) {
+          if (other == null) {
```
Comment (SonarTech) [inline src/SonarAnalyzer.Scanner/Protobuf/AnalyzerReport.cs]: MAJOR In member 'SymbolReferenceInfo.Types.SymbolReference.MergeFrom(SymbolReferenceInfo.Types.SymbolReference)', change parameter name 'other' to 'message' in order to match the identifier as it has been declared in 'IMessage<SymbolReferenceInfo.Types.SymbolReference>.MergeFrom(SymbolReferenceInfo.Types.SymbolReference)'. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=fxcop%3AParameterNamesShouldMatchBaseDeclaration)

File: src/SonarAnalyzer.Scanner/Protobuf/AnalyzerReport.cs [hunk 21]
```diff
@@ -0,0 +1,1835 @@
...
+      int size = 0;
+      if (FilePath.Length != 0) {
+        size += 1 + pb::CodedOutputStream.ComputeStringSize(FilePath);
+      }
+      size += issue_.CalculateSize(_repeated_issue_codec);
+      return size;
+    }
+
+    public void MergeFrom(FileIssues other) {
+      if (other == null) {
```
Comment (SonarTech) [inline src/SonarAnalyzer.Scanner/Protobuf/AnalyzerReport.cs]: MAJOR In member 'FileIssues.MergeFrom(FileIssues)', change parameter name 'other' to 'message' in order to match the identifier as it has been declared in 'IMessage<FileIssues>.MergeFrom(FileIssues)'. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=fxcop%3AParameterNamesShouldMatchBaseDeclaration)

**Comment by SonarTech** [comment]
SonarQube analysis reported 24 issues
* MAJOR 24 major

Watch the comments in this conversation to review them.

#### 2 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR sonaranalyzer-csharp-vbnet:sonaranalyzer-csharp-vbnet:79F0B359-8275-4F3B-B919-82C493B74665: Do not nest type 'TokenTypeInfo.Types'. Alternatively, change its accessibility so that it is not externally visible. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=fxcop%3ANestedTypesShouldNotBeVisible)
1. MAJOR sonaranalyzer-csharp-vbnet:sonaranalyzer-csharp-vbnet:79F0B359-8275-4F3B-B919-82C493B74665: Do not nest type 'TokenTypeInfo.Types.TokenInfo'. Alternatively, change its accessibility so that it is not externally visible. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=fxcop%3ANestedTypesShouldNotBeVisible)

File: src/SonarAnalyzer.Scanner/Program.cs [hunk 22]
```diff
@@ -19,24 +19,22 @@
...
 
 namespace SonarAnalyzer.Runner
 {
     public static class Program
     {
-        internal const string AnalysisOutputFileName = "analysis-output.xml";
-        internal const string TokenInfosFileName = "token-infos.dat";
-        internal const string TokenReferenceInfosFileName = "token-reference-infos.dat";
-        internal const string CopyPasteTokenInfosFileName = "token-cpd-infos.dat";
+        internal const string TokenTypeFileName = "token-type.dat";
```
Comment (henryju) [inline src/SonarAnalyzer.Scanner/Program.cs]: Naming the file syntax-highlighting.dat would be more consistent with SQ APIs.

Comment (henryju) [inline src/SonarAnalyzer.Scanner/Program.cs]: In platform we tend to name all Protobuf files with .pb suffix. This would help to know what is the format.

Comment (tvsonar) [inline src/SonarAnalyzer.Scanner/Program.cs]: The file itself only contains token type information, it's only the platform that decides to use this for syntax highlighting.

**Comment by SonarTech** [comment]
SonarQube analysis reported 24 issues
* MAJOR 24 major

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR sonaranalyzer-csharp-vbnet:sonaranalyzer-csharp-vbnet:79F0B359-8275-4F3B-B919-82C493B74665: D:\slave02\workspace\sonaranalyzer-dotnet-build\1b34ef03\src\SonarAnalyzer.Scanner\Protobuf\AnalyzerReport.cs line 24: Initialize all static fields in 'AnalyzerReportReflection' when those fields are declared and remove the explicit static constructor. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=fxcop%3AInitializeReferenceTypeStaticFieldsInline)
1. MAJOR sonaranalyzer-csharp-vbnet:sonaranalyzer-csharp-vbnet:79F0B359-8275-4F3B-B919-82C493B74665: D:\slave02\workspace\sonaranalyzer-dotnet-build\1b34ef03\src\SonarAnalyzer.Scanner\Protobuf\AnalyzerReport.cs line 856: In member 'CopyPasteTokenInfo.Equals(object)', change parameter name 'other' to 'obj' in order to match the identifier as it has been declared in 'object.Equals(object)'. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=fxcop%3AParameterNamesShouldMatchBaseDeclaration)
1. MAJOR sonaranalyzer-csharp-vbnet:sonaranalyzer-csharp-vbnet:79F0B359-8275-4F3B-B919-82C493B74665: D:\slave02\workspace\sonaranalyzer-dotnet-build\1b34ef03\src\SonarAnalyzer.Scanner\Protobuf\AnalyzerReport.cs line 900: In member 'CopyPasteTokenInfo.MergeFrom(CopyPasteTokenInfo)', change parameter name 'other' to 'message' in order to match the identifier as it has been declared in 'IMessage<CopyPasteTokenInfo>.MergeFrom(CopyPasteTokenInfo)'. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=fxcop%3AParameterNamesShouldMatchBaseDeclaration)
1. MAJOR sonaranalyzer-csharp-vbnet:sonaranalyzer-csharp-vbnet:79F0B359-8275-4F3B-B919-82C493B74665: D:\slave02\workspace\sonaranalyzer-dotnet-build\1b34ef03\src\SonarAnalyzer.Scanner\Protobuf\AnalyzerReport.cs line 981: In member 'CopyPasteTokenInfo.Types.TokenInfo.Equals(object)', change parameter name 'other' to 'obj' in order to match the identifier as it has been declared in 'object.Equals(object)'. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=fxcop%3AParameterNamesShouldMatchBaseDeclaration)
1. MAJOR sonaranalyzer-csharp-vbnet:sonaranalyzer-csharp-vbnet:79F0B359-8275-4F3B-B919-82C493B74665: D:\slave02\workspace\sonaranalyzer-dotnet-build\1b34ef03\src\SonarAnalyzer.Scanner\Protobuf\AnalyzerReport.cs line 1030: In member 'CopyPasteTokenInfo.Types.TokenInfo.MergeFrom(CopyPasteTokenInfo.Types.TokenInfo)', change parameter name 'other' to 'message' in order to match the identifier as it has been declared in 'IMessage<CopyPasteTokenInfo.Types.TokenInfo>.MergeFrom(CopyPasteTokenInfo.Types.TokenInfo)'. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=fxcop%3AParameterNamesShouldMatchBaseDeclaration)
1. MAJOR sonaranalyzer-csharp-vbnet:sonaranalyzer-csharp-vbnet:79F0B359-8275-4F3B-B919-82C493B74665: D:\slave02\workspace\sonaranalyzer-dotnet-build\1b34ef03\src\SonarAnalyzer.Scanner\Protobuf\AnalyzerReport.cs line 1587: In member 'FileIssues.Equals(object)', change parameter name 'other' to 'obj' in order to match the identifier as it has been declared in 'object.Equals(object)'. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=fxcop%3AParameterNamesShouldMatchBaseDeclaration)
1. MAJOR sonaranalyzer-csharp-vbnet:sonaranalyzer-csharp-vbnet:79F0B359-8275-4F3B-B919-82C493B74665: D:\slave02\workspace\sonaranalyzer-dotnet-build\1b34ef03\src\SonarAnalyzer.Scanner\Protobuf\AnalyzerReport.cs line 1631: In member 'FileIssues.MergeFrom(FileIssues)', change parameter name 'other' to 'message' in order to match the identifier as it has been declared in 'IMessage<FileIssues>.MergeFrom(FileIssues)'. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=fxcop%3AParameterNamesShouldMatchBaseDeclaration)
1. MAJOR sonaranalyzer-csharp-vbnet:sonaranalyzer-csharp-vbnet:79F0B359-8275-4F3B-B919-82C493B74665: D:\slave02\workspace\sonaranalyzer-dotnet-build\1b34ef03\src\SonarAnalyzer.Scanner\Protobuf\AnalyzerReport.cs line 1723: In member 'FileIssues.Types.Issue.Equals(object)', change parameter name 'other' to 'obj' in order to match the identifier as it has been declared in 'object.Equals(object)'. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=fxcop%3AParameterNamesShouldMatchBaseDeclaration)
1. MAJOR sonaranalyzer-csharp-vbnet:sonaranalyzer-csharp-vbnet:79F0B359-8275-4F3B-B919-82C493B74665: D:\slave02\workspace\sonaranalyzer-dotnet-build\1b34ef03\src\SonarAnalyzer.Scanner\Protobuf\AnalyzerReport.cs line 1781: In member 'FileIssues.Types.Issue.MergeFrom(FileIssues.Types.Issue)', change parameter name 'other' to 'message' in order to match the identifier as it has been declared in 'IMessage<FileIssues.Types.Issue>.MergeFrom(FileIssues.Types.Issue)'. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=fxcop%3AParameterNamesShouldMatchBaseDeclaration)
1. MAJOR sonaranalyzer-csharp-vbnet:sonaranalyzer-csharp-vbnet:79F0B359-8275-4F3B-B919-82C493B74665: D:\slave02\workspace\sonaranalyzer-dotnet-build\1b34ef03\src\SonarAnalyzer.Scanner\Protobuf\AnalyzerReport.cs line 1265: In member 'MetricsInfo.Equals(object)', change parameter name 'other' to 'obj' in order to match the identifier as it has been declared in 'object.Equals(object)'. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=fxcop%3AParameterNamesShouldMatchBaseDeclaration)

**Pull request merged**